### PR TITLE
Optimize SystemObject unit test 

### PR DESCRIPTION
**Pull request by yufengwangca**
#### Problem
What is being fixed?  Examples:
* SystemObject Unit Test take more than 10s to complete on my local Linux machine, and it does not test ObjectPool boundary condition since kPoolSize(122) is not set as a multiple of kNumThreads(16).

#### Change overview
* Reduce alloc/free iteration from 100000 to 1000 to make SystemObject unit test complete within 2 seconds.
* Set kPoolSize to 112 to be multiple of kNumThreads(16).

#### Testing
How was this tested? (at least one bullet point required)
* $ ./TestSystemObject 
```
'#0:','chip-system-object'
'#2:','Setup                   ','PASSED'
'#3:','Retention               ','PASSED'
'#3:','Concurrency             ','PASSED'
'#3:','HighWatermark           ','PASSED'
'#3:','HighWatermarkConcurrency','PASSED'
'#4:','Teardown                ','PASSED'
'#6:','0','4'
'#7:','0','83429'
```

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Review by mspang — APPROVED** [review]

File: src/system/tests/TestSystemObject.cpp [hunk 1]
```diff
@@ -80,15 +80,15 @@ class TestObject : public Object
 private:
     enum
     {
-        kPoolSize = 122 // a multiple of kNumThreads, less than CHIP_SYS_STATS_COUNT_MAX
+        kPoolSize = 112 // a multiple of kNumThreads, less than CHIP_SYS_STATS_COUNT_MAX
     };
     static ObjectPool<TestObject, kPoolSize> sPool;
```
Comment (bzbarsky-apple) [inline src/system/tests/TestSystemObject.cpp:85]: How about we add a static_assert that `kPoolSize % kNumThreads == 0`?  And perhaps that `kPoolSize < CHIP_SYS_STATS_COUNT_MAX`.

**Review by bzbarsky-apple — APPROVED** [review]

File: src/system/tests/TestSystemObject.cpp (hunk 1 above)
Comment (yufengwangca) [reply src/system/tests/TestSystemObject.cpp:85]: Acked

**Review by saurabhst — APPROVED** [review]
